### DISCUSSION TITLE
Is it possible to use components like Form and Combobox without a store prop?

### DISCUSSION THREAD

**Discussion by jhowell-droit** [ariakit/ariakit#2464] [Questions] [answered]
Hi,

I'm currently building a design system and am experimenting by using AriaKit to add some missing accessibility features to my components. I'm basically creating my own components and then instantiating the AriaKit components inside them like so:

```
const Form = ({ children, heading, store, ...rest }: FormComponentProps) => {
  const form = useFormStore({
    defaultValues: { firstName: '', lastName: '', country: '', city: '', cuisine: '', email: '' }
  });
  const id = uniqid();
  return (
    <AriaKitForm
      className={clsx({
        [styles['c-form']]: true
      })}
      aria-labelledby={id}
      store={form}
      {...rest}
    >
      <Text as="h2" id={id} color={1} weight="medium">
        {heading}
      </Text>
      {children}
    </AriaKitForm>
  );
};
```

My question is - is it possible to use these components without the `store` prop as ideally we'd like to manage the state of the Form (and other components) ourselves?

Also can we access the hooks that have been used in the components directly somehow as that may be another potential option?

**Comment by diegohaz** [comment, marked as answer]
> is it possible to use these components without the `store` prop as ideally we'd like to manage the state of the Form (and other components) ourselves?

The store is required, but you can manage the state yourself.

Example with React state:

```jsx
const [values, setValues] = useState(defaultValues)
const form = useFormStore({ values, setValues })
```

With props:

```jsx
function Form({ values, onChange }) {
  const form = useFormStore({ values, setValues: onChange })
}
```

More details on [component stores](https://ariakit.org/guide/component-stores).

> Also can we access the hooks that have been used in the components directly somehow as that may be another potential option?

Do you mean `useForm`, `useCombobox`, etc.? They're not exposed by the `@ariakit/react` library. You can import them from `@ariakit/react-core`, but this package won't be stable so soon.
